core/src/main/scala/aima/core/agent/SimpleProblemSolvingAgentProgram.scala

```diff
@@ -3,12 +3,12 @@ package aima.core.agent
 /**
   * @author Shawn Garner
   */
-trait SimpleProblemSolvingAgent[State, Action, Percept, Goal, Problem] extends Agent[Action, Percept] {
+trait SimpleProblemSolvingAgentProgram[PERCEPT, ACTION, STATE, GOAL, PROBLEM] extends AgentProgram[PERCEPT, ACTION] {
```

Comment:
@benderpremier would have been nicer if you had done the capitalization of the type parameters in a separate PR so this had only functional changes rather than cosmetics as well.